062-support-tiered-storage.md

```diff
@@ -0,0 +1,145 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively in Strimzi.
+
+## Motivation
+
+Supporting tiered storage in Strimzi would allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration of Kafka's `RemoteStorageManager` interface, Strimzi users would be able to easily adopt the tiered storage feature on Strimzi easily. More details about RemoteStorageManager can be found in [KIP405](https://cwiki.apache.org/confluence/display/KAFKA/KIP-405%3A+Kafka+Tiered+Storage).
+
+For simplicity, the proposal doesn't add support to customize `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation class `TopicBasedRemoteLogMetadataManager` in Kafka will be used.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+Add a `spec.kafka.tieredStorage` field in the `Kafka` CR, which would support configuration for all tiered storage related configuration.
+
+The `tieredStorage` config contains configuration for some selected configuration of tiered storage feature. Most importantly, the configuration for `RemoteStorageManager` is needed.
+
+```
+kafka:
+  tieredStorage:
+    type: customRemoteStorageManager
+    className: <ClassName>
+    classPath: <ClassPath>
+    config:
+      # A map with String keys and String values. Keys will be automatically prefixed with `rsm.config.`.
```

Comment:
we should have what we have in the example here ...

`# A map with String keys and String values. Key fields will be automatically prefixed with `rsm.config.` and appended to Kafka broker config.` 

because it's the first time we introduce the config part.

Reply:
Good catch, thanks!